### PR TITLE
feat(editor): Implement resource mapping for inputs in workflow tool

### PR DESCRIPTION
**Pull request by MiloradFilipovic**
## Summary
Adding new resource mapper to the Workflow tool as a way to specify sub-workflow inputs.

## Related Linear tickets, Github issues, and Community forum posts
Closes ADO-2910

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.node.ts [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+				//         source:database
+				// ----------------------------------
+				{
+					displayName: 'Workflow ID',
+					name: 'workflowId',
+					type: 'string',
+					displayOptions: {
+						show: {
+							source: ['database'],
+							'@version': [{ _cnd: { lte: 1.1 } }],
```
[the remark this answers is not in the document]
Comment (MiloradFilipovic) [reply packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.node.ts:172]: Yup, makes sense

File: packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.node.ts [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+							source: ['parameter'],
+						},
+					},
+					default: '\n\n\n\n\n\n\n\n\n',
+					required: true,
+					description: 'The workflow JSON code to execute',
+				},
+			],
+		};
+	}
```
[the remark this answers is not in the document]
Comment (MiloradFilipovic) [reply packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.node.ts:252]: Sure

File: packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.node.ts [hunk 3]
```diff
@@ -0,0 +1,503 @@
...
+					'Error during parsing of JSON Schema. \n ' + error,
+				);
+			}
+		} else {
+			tool = new DynamicTool(functionBase);
+		}
+		return {
+			response: tool,
+		};
+	}
```
[the remark this answers is not in the document]
Comment (MiloradFilipovic) [reply packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.node.ts:48]: Agree, let me try breaking it up a bit.

File: packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/FromAIParser.ts [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+					return JSON.stringify(mappedResults);
+				} catch (error) {
+					const nodeError = new NodeOperationError(this.ctx.getNode(), error as Error);
+					this.ctx.addOutputData(NodeConnectionType.AiTool, index, nodeError);
+					return 'Error during node execution: ' + nodeError.description;
+				}
+			},
+		});
+
+		return tool;
```
[the remark this answers is not in the document]
Comment (MiloradFilipovic) [reply packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/FromAIParser.ts:418]: I would rather keep this for now but agree that we should remove unused parts. Will do

File: packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.test.ts [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+			const tool = await service.createTool(toolParams);
+			const result = await tool.func('test query');
+
+			expect(result).toContain('There was an error');
+			expect(context.addOutputData).toHaveBeenCalled();
+		});
+	});
+
+	describe('handleToolResponse', () => {
```
Comment (OlegIvaniv) [inline packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/ToolWorkflowV2.test.ts:118]: <3

**Review by OlegIvaniv — APPROVED** [review]
Very nice, V2 is much easier to follow now. A kudos for adding tests!

**Pull request merged**